src/main/rx/fport.c

```diff
@@ -147,41 +147,38 @@ static void fportDataReceive(uint16_t c, void *data)
     UNUSED(data);
 
     static timeUs_t frameStartAt = 0;
-    static bool frameStarting = false;
     static bool escapedCharacter = false;
-    static uint8_t bufferPosition = 0;
     static timeUs_t lastFrameReceivedUs = 0;
     static bool telemetryFrame = false;
 
     const timeUs_t currentTimeUs = micros();
 
-    if (!skipUntilStart && (frameStarting || bufferPosition > 0) && cmpTimeUs(currentTimeUs, frameStartAt) > FPORT_TIME_NEEDED_PER_FRAME_US + 500) {
-         reportFrameError(DEBUG_FPORT_ERROR_TIMEOUT);
+    if (framePosition > 0) {
+        clearToSend = false;
```

Comment:
`clearToSend` will be always false here

But `clearToSend` should be cleared on any received byte except terminating FPORT_FRAME_MARKER of telemetry frame

Reply:
See below, except for the bytes that are feedback of the bytes we are sending.

Reply:
`framePosition == 0` when feedback bytes are received ...

Reply:
I still don't understand. You are saying that `clearToSend` will be always false here, but at the same time you are saying that it should be cleared...

Reply:
It will be false when `(framePosition > 0)`. But other byte than `FPORT_FRAME_MARKER` may be received between frames (noice, our reply, etc.) and `clearToSend` should be cleared in that case.

Reply:
This is what I am doing - except in the case where we are in between frames, since the sender should be able to check before every byte if it is still clear to send (it does not do this at the moment).

Reply:
Any unexpected byte received between frames should IMO reset `clearToSend`. Echoded response bytes may be handled same way when only one response frame is sent.

Sender may theoretically check that bytes are echoed back correctly (at character or frame level) to detect collision, but that will be quite complicated and will gain almost nothing.

Sender queue is buffered, checking between bytes will need modification of UART driver. Again it is not necessary and impractical.